packages/wazuh_testing/wazuh_testing/fim.py

```diff
@@ -394,53 +400,73 @@ def validate_checkers_per_event():
-        assert (event_types[ev_type] == n_regular)
+        assert (event_types[ev_type] == len(file_list))
 
     def check_files_in_event():
         file_paths = jq(".[].data.path").transform(events, multiple_output=True)
-        for file_name in range(n_regular):
-            assert (os.path.join(folder, f'regular_{file_name}') in file_paths)
+        for file_name, file_content in file_list.items():
+            assert (os.path.join(folder, file_name) in file_paths)
+
+    def check_events(event_type, validate_after):
+        if events is not None:
+            validate_checkers_per_event()
+            check_events_type(event_type)
+            check_files_in_event()
+            run_custom_validators(validators_after_cud)
+            run_custom_validators(validate_after)
 
     def run_custom_validators(validators):
         if validators is not None:
             for validator in validators:
                 for event in events:
                     validator(event)
 
+    # Transform file list
+    if isinstance(file_list, list):
+        aux_dict = {i: '' for i in file_list}
+        file_list = aux_dict
+    elif isinstance(file_list, dict):
+        pass
+    else:
+        raise Exception('Type error. It can only be list or dict')
+
     # Create text files
-    for name in range(n_regular):
-        create_file(REGULAR, f'regular_{name}', folder, '')
+    for name, content in file_list.items():
+        create_file(REGULAR, name, folder, content)
 
     check_time_travel()
     events = fetch_events()
-    if events is not None:
-        validate_checkers_per_event()
-        check_events_type('added')
-        check_files_in_event()
-        run_custom_validators(validators_after_cud)
-        run_custom_validators(validators_after_create)
+    check_events('added', validators_after_create)
 
     # Modify previous text files
-    for name in range(n_regular):
-        modify_file(folder, f'regular_{name}', 'Sample content added')
+    for name in file_list:
+        modify_file(folder, name, 'Sample content added')
 
     check_time_travel()
     events = fetch_events()
-    if events is not None:
-        validate_checkers_per_event()
-        check_events_type('modified')
-        check_files_in_event()
-        run_custom_validators(validators_after_cud)
-        run_custom_validators(validators_after_update)
+    check_events('modified', validators_after_update)
 
     # Delete previous text files
-    for name in range(n_regular):
-        delete_file(folder, f'regular_{name}')
+    for name in file_list:
+        delete_file(folder, name)
 
     check_time_travel()
     events = fetch_events()
-    if events is not None:
-        check_events_type('deleted')
-        check_files_in_event()
-        run_custom_validators(validators_after_cud)
-        run_custom_validators(validators_after_delete)
+    check_events('deleted', validators_after_delete)
+
+
+def restart_wazuh_with_new_conf(new_conf, log_monitor):
```

Comment:
It is ambiguous with `restart_wazuh` and `wait_for_initial_scan` fixtures. Need refactor.

Reply:
Done